server/xpub-controller/helpers/manuscript.js

```diff
@@ -0,0 +1,79 @@
+const logger = require('@pubsweet/logger')
+const FilesHelper = require('./files')
+const FileModel = require('@elifesciences/xpub-model').File
+const ManuscriptModel = require('@elifesciences/xpub-model').Manuscript
+
+class ManuscriptHelper {
+  constructor(config, userId, storage, filesHelper) {
+    this.userId = userId
+    this.config = config
+    this.storage = storage
+    this.filesHelper = filesHelper
+  }
+
+  async uploadManuscriptFile(fileData, fileSize, manuscriptId, progress) {
+    return new Promise(async (resolve, reject) => {
+      const { stream } = fileData
+      let { fileEntity } = fileData
+      const { id: fileId, filename, mimeType } = fileEntity
+
+      logger.info(
+        `Manuscript Upload Size: ${filename}, ${fileSize} | ${manuscriptId}`,
+      )
+
+      logger.info(
+        `Manuscript Upload fileContents::start ${filename} | ${manuscriptId}`,
+      )
+      const fileContent = await FilesHelper.uploadFileToServer(stream, fileSize)
+
+      fileEntity = await FileModel.find(fileId)
+      await fileEntity.updateStatus('UPLOADED')
+
+      logger.info(
+        `Manuscript Upload fileContents::end ${filename} | ${manuscriptId}`,
+      )
+
+      logger.info(`Manuscript Upload S3::start ${filename} | ${manuscriptId}`)
+
+      fileEntity = await FileModel.find(fileId)
+
+      try {
+        await this.storage.putContent(fileEntity, fileContent, {
+          size: fileSize,
+        })
+        await fileEntity.updateStatus('STORED')
+      } catch (err) {
+        logger.error(
+          `Manuscript was not uploaded to S3: ${err} | ${manuscriptId}`,
+        )
+        await fileEntity.delete()
+        FilesHelper.endFileProgress(progress)
```

Comment:
Just a note for #1521 this should probably be removed from this helper function (which would mean no longer requiring progress as a param). The reject will then be handled at the controller level to end file progress.